hypershift-operator/controllers/sharedingress/router.go

```diff
@@ -145,6 +144,7 @@ func ReconcileRouterConfiguration(cm *corev1.ConfigMap, config string) error {
 
 func ReconcileRouterDeployment(deployment *appsv1.Deployment, config *corev1.ConfigMap) error {
 	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas: ptr.To(int32(2)),
```

Comment:
this deployment should also have affinity preferences to distribute the replicas across Zones/Nodes.
Similar to what we already do with setMultizoneSpread/setNodeSpread for each HCP pods

Reply:
done